pyodide_build/buildall.py

```diff
@@ -649,22 +650,18 @@ def run(self, n_jobs: int, already_built: set[str]) -> None:
 
 
 def _ensure_rust_toolchain():
-    rust_toolchain = build_env.get_build_flag("RUST_TOOLCHAIN")
+    toolchain_version = build_env.get_build_flag("RUST_TOOLCHAIN")
+    pyodide_root = get_pyodide_root()
+    toolchain_path = pyodide_root / ".rust-toolchain" / toolchain_version
+    if not toolchain_path.exists():
+        download_and_unpack_archive(
+            f"http://pyodide-cache.s3-website-us-east-1.amazonaws.com/rustc/{toolchain_version}.tar.bz2",
+            toolchain_path.parent,
+            "rust toolchain",
+        )
```

Comment:
Lets add a configuration variable for this, and probably make it opt-in to download the toolchain from the custom URL.

For instance:

```python
rust_toolchain: "nightly-2024-01-29"
rust_toolchain_url_base: "" 
```

```python
toolchain_version = build_env.get_build_flag("RUST_TOOLCHAIN")
toolchain_url = build_env.get_build_flag("RUST_TOOLCHAIN_URL_BASE")

if toolchain_url:
    # if custom toolchain URL is given, download from it
else:
    # otherwise, download the standard toolchain
```

so we can,

1) change the URL without updating the pyodide-build version
2) make this feature optional, so people can still build packages against older pyodide versions easily

Reply:
> maybe we can check for ABI 2025_0 or later? Because if you opt out you won't get an ABI compatible wheel.

Makes sense. Otherwise, if you think branching out using the ABI version will make the code too complex and increase the maintenance burden, I am okay with dropping support for Pyodide <= 0.27.X. We have [`min_pyodide_build_version` and `max_pyodide_build_version`](https://github.com/pyodide/pyodide/blob/main/pyodide-cross-build-environments.json) for that purpose.